x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { EuiBasicTableColumn, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { TimeRange } from '@kbn/es-query';
+
+import { useKibanaContextForPlugin } from '../../../../hooks/use_kibana';
+import { createInventoryMetricFormatter } from '../../inventory_view/lib/create_inventory_metric_formatter';
+import { HostsTableEntryTitle } from '../components/hosts_table_entry_title';
+import type {
+  SnapshotNode,
+  SnapshotNodeMetric,
+  SnapshotMetricInput,
+} from '../../../../../common/http_api';
+
+/**
+ * Columns and items types
+ */
+export type CloudProvider = 'gcp' | 'aws' | 'azure' | 'unknownProvider';
+
+type HostMetric = 'cpuCores' | 'diskLatency' | 'rx' | 'tx' | 'memory' | 'memoryTotal';
+
+type HostMetrics = Record<HostMetric, SnapshotNodeMetric>;
+
+export interface HostNodeRow extends HostMetrics {
+  os?: string | null;
+  servicesOnHost?: number | null;
+  title: { name: string; cloudProvider?: CloudProvider | null };
+  name: string;
+}
+
+// type MappedMetrics = Record<keyof HostNodeRow, SnapshotNodeMetric>;
+
+interface HostTableParams {
+  time: TimeRange;
+}
+
+/**
+ * Helper functions
+ */
+const formatMetric = (type: SnapshotMetricInput['type'], value: number | undefined | null) => {
+  return value || value === 0 ? createInventoryMetricFormatter({ type })(value) : 'N/A';
+};
+
+const buildItemsList = (nodes: SnapshotNode[]) => {
+  return nodes.map(({ metrics, path, name }) => ({
+    name,
+    os: path.at(-1)?.os ?? '-',
+    title: {
+      name,
+      cloudProvider: path.at(-1)?.cloudProvider ?? null,
+    },
+    ...metrics.reduce((data, metric) => {
+      data[metric.name as HostMetric] = metric;
+      return data;
+    }, {} as HostMetrics),
+  })) as HostNodeRow[];
+};
+
+/**
+ * Columns translations
+ */
+const titleLabel = i18n.translate('xpack.infra.hostsTable.nameColumnHeader', {
```

Comment:
Out of curiosity: why did you separate the translations from the columns array?

Reply:
Being those translations part of the `columns` list, every time the `time` updates it implies recomputing those same static translations invoking the `i18n.translate()` function for each column.

Moving those static texts out of the functions, they are computed once at module time and we'll save various steps in the call stack on each column recomputing.

I myself consider this a micro-optimization, but it's worth leaving it clean and split :) Let me know your thoughts and if you see it better in the function.

Reply:
Yeah, it makes sense. I like it. I was thinking more about the trade-off between readability and actual gain from the optimization, because the total nr of lines roughly doubled after it. But the file is still small, so I wouldn't worry about it now.